homeassistant/components/voip/assist_satellite.py

```diff
@@ -165,31 +167,36 @@ def on_chunk(self, audio_bytes: bytes) -> None:
                 "voip_pipeline_run",
             )
 
+        _LOGGER.debug("Putting audio bytes on queue")
         self._audio_queue.put_nowait(audio_bytes)
 
+    async def _stt_stream(self) -> AsyncIterable[bytes]:
+        while True:
+            async with asyncio.timeout(self._audio_chunk_timeout):
+                chunk = await self._audio_queue.get()
+            if not chunk:
```

Comment:
Ah I see, the change is to not put this inside the timeout `with` block. But how does that fix it? It means the chunk is set to the previous value, which is true, otherwise it would have broken.

It means that we will also yield the same audio chunk twice, which is not something we would want.